Source/JavaScriptCore/parser/Parser.h

```diff
@@ -1948,6 +1956,38 @@ class Parser {
         return nullptr;
     }
 
+    ALWAYS_INLINE bool findClosestFunctionScopeUntilStaticBlock()
```

Comment:
This is called in the hot code and it is not great since this is too costly. Can we have this information in each scope?
For example, if the upper scope has this flag, then the underlying scope should inherit it, unless it is function boundary. We should have a bool flag in scope which says this information, which is configured at the construction of the scope.